Google.PowerShell/Common/GcloudCmdlet.cs

```diff
@@ -87,9 +88,33 @@ protected string GetFullPath(string filePath)
                 return filePath;
             }
 
-            // For relative files, qualify them based on where the PowerShell session is.
-            string currentDirectory = SessionState.Path.CurrentLocation.Path;
-            return Path.Combine(currentDirectory, filePath);
+            ProviderInfo provider = null;
+            string[] result;
+
+            // resolve filepath and get the first one
+            try
+            {
+                result = GetResolvedProviderPathFromPSPath(filePath, out provider).ToArray();
+            }
+            catch (ItemNotFoundException itemEx)
+            {
+                // in case the file path is not created but we should still return the resolve path
+                result = new string[] { itemEx.ItemName };
+            }
+
+            if (result.Length != 1)
+            {
+                // cannot really resolve this
+                return null;
```

Comment:
I don't think this method should ever return null, right? If you ask to resolve a path (e.g. "foo/../bar") the worst thing you can do is just return "foo/../bar". Potentially returning null means every caller needs to check if the result is null, which greatly complicates things.

So an error mode could just be "do nothing and hope it works out".